Monika After Story/game/script-topics.rpy

```diff
@@ -2929,21 +2929,33 @@ init 5 python:
     addEvent(Event(persistent.event_database,eventlabel="monika_credits_song",category=['ddlc','media'],prompt="Credits song",random=True))
 
 label monika_credits_song:
-    #TODO: adjust based on starting act, doesn't quite apply to those that never went past act 3
-    m 1hua "I hope you liked my song."
-    m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
-    m 1eua "Give me some time, and I'll try to write another."
-    if persistent.instrument is not False:
-        if persistent.instrument:
-            m 3eua "Maybe you could play me a song too!"
+    if persistent.monika_kill:
+        m 1hua "I hope you liked my song."
+        m 1eka "I worked really hard on it. I know I'm not perfect at the piano yet, but I just couldn't let you go without telling you how I honestly felt about you."
```

Comment:
As a result of the previously mentioned bit, this line would need an alternate variant `if persistent.monika_kill`. I'm thinking something along the lines of:

`"I worked really hard on it. I know I'm not perfect at the piano yet, but I just wouldn't have been able to let you go without telling how I honestly felt about you."`

You can use a placeholder variable to substitute the pieces into this.

Reply:
That would basically make it so that players can only see the alternate topic once and than stick to the slightly altered original, that defeats the purpose of having an entire new branch for those who dont have persistent.monika_kill.

Reply:
That's kinda the idea. The alternate flow seems more like a first time introduction to the song itself rather than making sense on subsequent visits to this topic, especially after having heard it once within MAS (since you can, it's in the music menu)

Reply:
I will try to find some new dialogue than, though its sad that people will only see this once, especially when Moni repeats most topics word for word

Reply:
It's not once, it's until they listen to Your Reality.

Reply:
I know, but than it will be gone forever.

Reply:
I would like people who want to see it again to have the option to.

Reply:
That's fair, but just a slight FYI that a fair few topics and greetings do this, so it's not out of the norm

Reply:
I know, and I am sad I can never see those either, but anyway "I worked really hard on it. I know I'm not perfect at the piano yet, but I just really wanted to express how I feel about you." Will this do?

Reply:
Bit of a tense issue there. `but I just really wanted to express` is in past tense while `how I feel about you.` is in present.

Reply:
That should not be an issue, she expressed how she feels, indicates the feelings are still present.  She still loves you. "But I just really wanted to express how I felt about you". Indicates the feelings are gone.

Reply:
I personally disagree, it sounds off to me. And from "how I felt about you", I get "how I felt about you before" which isn't necessarily a bad thing, the feelings she felt before may be even stronger now.

Reply:
hmm ok, how about "I worked really hard on it. I know I'm not perfect at the piano yet, but I just really wanted to express my feelings towards you."

Reply:
ok using that

Reply:
Yeah I like that